### PR TITLE
feat(823): [4] release and tag trigger

### PR DESCRIPTION
**Pull request by tk3fftk**
This PR provides three things:
- https://github.com/screwdriver-cd/scm-github/pull/125/commits/7dbada6fff7ad7fe27d3acfd18b764bb2d2a5d30 : Enable to parse `release` and `tag` webhook
- https://github.com/screwdriver-cd/scm-github/pull/125/commits/aa56fced819e20371528f0f4f96df67a7a19207e : Implement  `getCommitRefSha`. It's used for getting sha when release and tag triggers are fired.
- https://github.com/screwdriver-cd/scm-github/pull/125/commits/e989a219a9e7e4f7e4975a6ae3c620d9558d2f3b : Add `create` and `release` event to GitHub hook permission.

Related Issue: https://github.com/screwdriver-cd/screwdriver/issues/823
Blocked by:
- https://github.com/screwdriver-cd/data-schema/pull/322
- https://github.com/screwdriver-cd/scm-base/pull/68
- https://github.com/screwdriver-cd/scm-router/pull/17

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1012,9 +1043,45 @@ class GithubScm extends Scm {
...
+                hookId,
+                scmContext: scmContexts[0],
+                ref: hoek.reach(webhookPayload, 'release.tag_name')
+            };
+        }
+        case 'create': {
+            const refType = hoek.reach(webhookPayload, 'ref_type');
+
+            if (refType !== 'tag') {
+                winston.info('A %s event of %s is not parsable.', type, refType);
```
Comment (wdstar) [inline index.js]: How about `A %s event of %s is not available yet in Screwdriver.cd.` ?

Comment (tk3fftk) [reply index.js]: It looks good suggestion. I think `in Screwdriver.cd.`  has no meaning so I'll fix with `in scm-github plugin`.

Comment (wdstar) [reply index.js]: 👍

File: index.js [hunk 2]
```diff
@@ -1012,9 +1043,45 @@ class GithubScm extends Scm {
...
+                type: 'repo',
+                username: hoek.reach(webhookPayload, 'sender.login'),
+                hookId,
+                scmContext: scmContexts[0],
+                ref: hoek.reach(webhookPayload, 'ref')
+            };
+        }
+
         default:
+            winston.info('%s event is not parsable.', type);
```
Comment (wdstar) [inline index.js]: Same here. `%s event is not available yet in scm-github plugin`

Comment (tk3fftk) [reply index.js]: fixed!

**Pull request merged**